src/app/api/services/teaching-period.service.ts

```diff
@@ -23,42 +22,38 @@ export class TeachingPeriodService extends CachedEntityService<TeachingPeriod> {
       {
         keys: 'startDate',
         toEntityFn: MappingFunctions.mapDateToDay,
-        toJsonFn: MappingFunctions.mapDayToJson
+        toJsonFn: MappingFunctions.mapDayToJson,
       },
       {
         keys: 'endDate',
         toEntityFn: MappingFunctions.mapDateToDay,
-        toJsonFn: MappingFunctions.mapDayToJson
+        toJsonFn: MappingFunctions.mapDayToJson,
       },
       'activeUntil',
       'active',
       {
         keys: 'breaks',
         toEntityOp: (data, key, entity) => {
-          data['breaks']?.forEach(breakJson => {
+          data['breaks']?.forEach((breakJson) => {
             const teachingPeriod = entity as TeachingPeriod;
             const breakEntity = this.teachingPeriodBreakService.buildInstance(breakJson);
             teachingPeriod.breaksCache.add(breakEntity);
           });
-        }
+        },
       },
       {
         keys: 'units',
         toEntityOp: (data, key, entity) => {
-          data[key]?.forEach(unitJson => {
+          data[key]?.forEach((unitJson) => {
             const unitService: UnitService = AppInjector.get(UnitService);
-            const unit = unitService.cache.getOrCreate(
-              unitJson['id'],
-              unitService,
-              unitJson
-            );
+            const unit = unitService.cache.getOrCreate(unitJson['id'], unitService, unitJson);
             entity.unitsCache.add(unit);
           });
-        }
+        },
       }
     );
 
-    this.mapping.mapAllKeysToJsonExcept('id', 'unit', 'breaks');
+    this.mapping.mapAllKeysToJsonExcept('id', 'units', 'breaks');
```

Comment:
@macite - I think this was the issue, please review

Reply:
Agreed - this should avoid the issue.